gateway_tests/process/test_enum_undefined_timestamp.py

```diff
@@ -37,22 +57,49 @@ def on_change_ioc(pvname=None, value=None, timestamp=None, **kwargs):
             timestamp_to_string(timestamp)
         )
 
-    ioc_pv, gateway_pv = conftest.get_pv_pair(
-        "HUGO:ENUM",
-        auto_monitor=epics.dbr.DBE_PROPERTY,
-        ioc_callback=on_change_ioc,
-        gateway_callback=on_change_gateway,
-    )
-
-    ioc_value = ioc_pv.get()
-    gateway_value = gateway_pv.get()
-
-    # Verify timestamp and value match
-    assert ioc_value == gateway_value, f"ioc = {ioc_value} != gw = {gateway_value}"
-
-    # Now get the gateway value again and make sure the timestamp is not undefined
-    gateway_pv.get()
-    if ioc_pv.status != epics.dbr.AlarmStatus.UDF:
-        assert gateway_pv.status != epics.dbr.AlarmStatus.UDF, "2nd CA get is undefined!"
-    assert gateway_pv.timestamp != 0, "2nd CA get timestamp is undefined!"
-    assert ioc_value == gateway_value
+    with conftest.ca_subscription_pair(
+        "HUGO:ENUM", ioc_callback=on_change_ioc, gateway_callback=on_change_gateway,
+        mask=subscription_mask,
+    ) as (ioc_ch, gateway_ch):
+        for iteration in range(1, 5):
+            logger.info("Iteration %d", iteration)
+            ioc_md = ca.get_with_metadata(ioc_ch, ftype=dbr.TIME_ENUM)
+            gateway_md = ca.get_with_metadata(gateway_ch, ftype=dbr.TIME_ENUM)
+            assert ioc_md == gateway_md
+
+            if ioc_md["status"] != dbr.AlarmStatus.UDF:
+                assert gateway_md["status"] != dbr.AlarmStatus.UDF, "2nd CA get is undefined!"
+            assert gateway_md["timestamp"] != 0, "2nd CA get timestamp is undefined!"
+            assert ioc_md["value"] == gateway_md["value"]
+
+            time.sleep(0.1)
+
+
+@conftest.standard_test_environment_decorator
+def test_undefined_timestamp_get_only():
+    """
+    caget on an mbbi - without subscription configured.
```

Comment:
I can't help but think there must a be a good way to parameterize this and check other types too- but maybe there isn't.

Reply:
I think this could be - and should be done. Even if we have only reportedly seen this on ENUMs, it doesn't mean there won't be a regression in the future. I'll make an issue for a follow-up PR.